src/test/java/com/scopely/mapper/HashAndRange.java

```diff
@@ -1,17 +1,22 @@
 package com.scopely.mapper;
 
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAutoGeneratedKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
 import org.inferred.freebuilder.FreeBuilder;
+import org.jetbrains.annotations.Nullable;
 
 @FreeBuilder
 @JsonDeserialize(builder = HashAndRange.Builder.class)
 @DynamoDBTable(tableName = "hash_and_range")
 interface HashAndRange {
     @DynamoDBHashKey(attributeName = "hashKey")
     String getHashKey();
+    @Nullable
```

Comment:
We would want to get optional field's type parameter in order to check that
it is a string- but we can't get the actual type parameter at runtime.

On Sat, Aug 27, 2016, 23:26 Cristian <notifications@github.com> wrote:

> In src/test/java/com/scopely/mapper/HashAndRange.java
> <https://github.com/scopely/mapper/pull/2#discussion_r76526796>:
>
> >
> >  @FreeBuilder
> >  @JsonDeserialize(builder = HashAndRange.Builder.class)
> >  @DynamoDBTable(tableName = "hash_and_range")
> >  interface HashAndRange {
> >      @DynamoDBHashKey(attributeName = "hashKey")
> >      String getHashKey();
> > +    @Nullable
>
> but then there's no way we can check that we have a legal type for an
> autogenerated key at runtime
>
> How so?
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/scopely/mapper/pull/2/files/efba6e88619d0723a5e50cd1c811f2378c31da57#r76526796>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANr0F1vG2aDMDF0wPC7jZ5h2o5am_4Aks5qkSoEgaJpZM4JtoKg>
> .
>